npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -222,6 +225,33 @@ def updated_monorepo_dependencies
           updates
         end
 
+        def types_package
+          types_package_name = PackageName.new(dependency.name).types_package_name.to_s
+          top_level_dependencies.find { |d| d.name == types_package_name }
```

Comment:
You could memoize this to avoid computation on each call.

```suggestion
          @types_package ||= begin
            types_package_name = PackageName.new(dependency.name).types_package_name.to_s
            top_level_dependencies.find { |d| d.name == types_package_name }
          end
```